sonar-batch/src/main/java/org/sonar/batch/issue/tracking/LocalIssueTracking.java

```diff
@@ -214,46 +188,30 @@ private void addUnmatchedFromServer(Collection<ServerIssue> unmatchedIssues, Sou
     }
   }
 
-  private void addIssuesOnDeletedComponents(Collection<DefaultIssue> issues) {
+  private void addIssuesOnDeletedComponents(Collection<TrackedIssue> issues) {
     for (org.sonar.batch.protocol.input.BatchInput.ServerIssue previous : serverIssueRepository.issuesOnMissingComponents()) {
-      DefaultIssue dead = toUnmatchedIssue(previous);
+      TrackedIssue dead = IssueTransformer.toTrackedIssue(previous);
       updateUnmatchedIssue(dead, true);
       issues.add(dead);
     }
   }
 
-  private DefaultIssue toUnmatchedIssue(org.sonar.batch.protocol.input.BatchInput.ServerIssue serverIssue) {
-    DefaultIssue issue = new DefaultIssue();
-    issue.setKey(serverIssue.getKey());
-    issue.setStatus(serverIssue.getStatus());
-    issue.setResolution(serverIssue.hasResolution() ? serverIssue.getResolution() : null);
-    issue.setMessage(serverIssue.hasMsg() ? serverIssue.getMsg() : null);
-    issue.setLine(serverIssue.hasLine() ? serverIssue.getLine() : null);
-    issue.setSeverity(serverIssue.getSeverity().name());
-    issue.setAssignee(serverIssue.hasAssigneeLogin() ? serverIssue.getAssigneeLogin() : null);
-    issue.setComponentKey(ComponentKeys.createEffectiveKey(serverIssue.getModuleKey(), serverIssue.hasPath() ? serverIssue.getPath() : null));
-    issue.setManualSeverity(serverIssue.getManualSeverity());
-    issue.setCreationDate(new Date(serverIssue.getCreationDate()));
-    issue.setRuleKey(RuleKey.of(serverIssue.getRuleRepository(), serverIssue.getRuleKey()));
-    issue.setNew(false);
-    return issue;
-  }
-
-  private void updateUnmatchedIssue(DefaultIssue issue, boolean forceEndOfLife) {
+  private void updateUnmatchedIssue(TrackedIssue issue, boolean forceEndOfLife) {
     ActiveRule activeRule = activeRules.find(issue.ruleKey());
     issue.setNew(false);
 
     boolean manualIssue = issue.ruleKey().isManual();
     boolean isRemovedRule = activeRule == null;
-    if (manualIssue) {
-      issue.setBeingClosed(forceEndOfLife || isRemovedRule);
-    } else {
-      issue.setBeingClosed(true);
+
+    if (isRemovedRule) {
+      IssueTransformer.resolveRemove(issue);
+    }
+    else if (forceEndOfLife || !manualIssue) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this "else" on the same line that the previous closing curly brace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)